test/functional/mempool_package_limits.py

```diff
@@ -3,20 +3,39 @@
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test logic for limiting mempool and package ancestors/descendants."""
-
-from decimal import Decimal
-
 from test_framework.blocktools import COINBASE_MATURITY
-from test_framework.test_framework import BitcoinTestFramework
 from test_framework.messages import (
-    COIN,
     WITNESS_SCALE_FACTOR,
 )
+from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import (
     assert_equal,
 )
 from test_framework.wallet import MiniWallet
 
+
+# Decorator to check that
+# 1) mempool is empty at the start of a subtest
+# 2) after running the subtest, submitting the created package
+#    fails with the error "package-mempool-limits" for each tx
+# 3) after mining a block, submitting the created package succeeds
```

Comment:
(1) I think it could be useful to clarify that the subtest modifies mempool and creates the txns
(2) I prefer to reserve the term "submit" for sendraw and submitpackage

```suggestion
# 2) run the subtest, which may submit some transaction(s) to the mempool and create a list of hex transactions
# 3) testmempoolaccept the package hex and check that it fails with the error "package-mempool-limits" for each tx
# 4) after mining a block, clearing the pre-submitted transactions from mempool, submitting the created package succeeds
```

Reply:
Thanks, updated the description! (Makes sense to use "submit" only for sendrawtransaction and submitpackage RPCs; neat idea of using "testmempoolaccept" directly as a verb :)).